### PR TITLE
[TheiaProk] expose sistr optional param inputs to theiaProk wfs

### LINKED ISSUE #376: expose sistr optional inputs to TheiaProk wfs
:cool:

### :pushpin: Explain the Request

This is low priority, but I just realized that the user does not have the ability to change SISTR optional inputs for the various TheiaProk workflows since the task is within the merlin_magic subworkflow.

### :books: Context

Low priority. Not requested by any users AFAIK

### :chart_with_upwards_trend: Desired Behavior

Allow user to change optional inputs for SISTR, namely cpu, memory, docker, and the `Boolean use_full_cgmlst_db` optional input: https://github.com/theiagen/public_health_bioinformatics/blob/c3f3b705bf20349dbec54567442b7ed28fe2a93a/tasks/species_typing/task_sistr.wdl#L16

### PR DESCRIPTION
**Pull request by fraser-combe**
This PR closes #376

🗑️ This dev branch should be deleted after merging to main.

## :brain: Summary
This allows the ability to expose the parameters cpu, memory, docker image and use_full_cmlst_db (boolean) as optional outputs from magic_merlin to SISTR task

## :zap: Impacted Workflows/Tasks
The PR only exposes the optional parameters for the TheiaProk workflows that contain salmonella data in the Magic_merlin workflow that calls SISTR (Salmonella In Silico Typing Resource) task.

This PR may lead to different results in pre-existing outputs: No

This PR uses an element that could cause duplicate runs to have different results: No

## :hammer_and_wrench: Changes
Added optional parameter outputs to Magic merlin

### :gear: Algorithm
None only optional parameters exposed to SISTR task

### ➡️ Inputs
cpu, memory, docker, and the Boolean use_full_cgmlst_db are now optional inputs

### ⬅️ Outputs
No changes

## :test_tube: Testing
Salmonella testing data has been run through TheiaProk_Illumina_PE_PHB changing optional parameters - script successfully updated with optional parameter changes in output files and run was successful

https://app.terra.bio/#workspaces/theiagen-training-workspaces/Theiagen_Combe_Sandbox/job_history/563e977b-3cd5-4e1a-b0c6-6de15fafe425

## :microscope: Final Developer Checklist

- [X ] The workflow/task has been tested and results, including file contents, are as anticipated
- [X] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [X] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [x] All changed results have been confirmed
- [x] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [x] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments
- [ ] The documentation has been updated

### CODE REVIEW

**Comment by sage-wright** [comment]
Looks great, @fraser-combe! Will you add these new optional inputs to the documentation? You'll have to pull main into your branch and it should be a quick addition!

**Comment by fraser-combe** [comment]
Merged with main and updated docs to display the new SISTR outputs

File: docs/workflows/genomic_characterization/theiaprok.md [hunk 1]
```diff
@@ -372,6 +372,9 @@ All input reads are processed through "[core tasks](#core-tasks-performed-for-al
 | merlin_magic | **shigatyper_docker_image** | String | The Docker container to use for the task | us-docker.pkg.dev/general-theiagen/staphb/shigatyper:2.0.5 | Optional | FASTA, ONT, PE, SE |
 | merlin_magic | **shigeifinder_docker_image** | String | The Docker container to use for the task | us-docker.pkg.dev/general-theiagen/staphb/shigeifinder:1.3.5 | Optional | FASTA, ONT, PE, SE |
 | merlin_magic | **sistr_docker_image** | String | The Docker container to use for the task | us-docker.pkg.dev/general-theiagen/biocontainers/sistr_cmd:1.1.1--pyh864c0ab_2 | Optional | FASTA, ONT, PE, SE |
+| merlin_magic | **sistr_cpu** | Int | The number of CPU cores to allocate for the task. | 8 | Optional | FASTA, ONT, PE, SE  |
```
Comment (sage-wright) [inline docs/workflows/genomic_characterization/theiaprok.md:375]: super nit-picky: can you order these outputs in alphabetical order with the surrounding? 

our ordering priority is 1) 1st column, then 2) 2nd column

**Comment by sage-wright** [comment]
@fraser-combe, I'll merge once the tests are passing! Looks like a couple md5sums got out of date when you merged in main

**Review by sage-wright — APPROVED** [review]
⭐ well done!

**Pull request merged**